openml/datasets/functions.py

```diff
@@ -418,22 +416,62 @@ def create_dataset(name, description, creator, contributor, collection_date,
         'data': data
     }
 
+    # Determine arff format from the dataset
```

Comment:
move this block before building the `arff_object`

Reply:
True, this way if there is a problem with the data we skip the object creation and gc.